### ISSUE TITLE
Julia block comments

### ISSUE THREAD

**Issue by elextr** [geany/geany#3368]
Should Julia be mentioned in https://github.com/geany/geany/blob/607fcec1fa5aff005090fbda17280976dcee68c8/src/editor.c#L3431?

**Comment by techee** [comment]
I think so. Also, the list of languages seems way too short and I suspect many more are missing.

**Comment by techee** [comment]
Now after looking at #3386 where the `in_block_comment()` is used, it's sole purpose is to tell whether to add the initial `*` in multi-line block comment. This now works only for `*` and `+` for D but no other character. Block comments in Julia are introduced by `#=` so we'd have to do the equivalent for the `=` character too and I'm not really sure if Julia users write comments this way:
```Julia
#= line1
 = line2
 = line3 =#
```

or rather just

```Julia
#= line1
   line2
   line3 =#
```

and whether it's something they want.

**Comment by elextr** [comment]
I just mentioned Julia because its where I noticed it, really the issue is a prompt to check all languages that are not mentioned in the switch (only 8 of Geanys 30+, perhaps others should be listed in a comment after the `default:` so its clear their omission is deliberate)

Specifically for Julia, AFAICT multi-line comments are only used for commenting out blocks of code, it uses doc strings like python, so multi-line comments are not used for documentation purposes, so probably ok to omit.

**Comment by ntrel** [comment]
> the list of languages seems way too short and I suspect many more are missing.

Maybe, though SCLEX_CPP is used for various file types. Possibly other lexers are reused too.

**Comment by elextr** [comment]
@ntrel Good point, the best solution would be if the list of comment styles came from the filetype file so custom styles could alter it, but "somebodys" gotta do it[^1] and since @techee says its only used for comment continuation maybe its not worth it.

[^1]: :tm: @eht16

**Comment by ntrel** [comment]
> if the list of comment styles came from the filetype file so custom styles could alter it

AFAIK the styles are just compile-time constants so it's not obvious how they could be stored in the filetype file. Perhaps Scintilla provides some other way too, IDK.

More generally it would be good if we could auto-generate C code based on scintilla styles so adding a lexer is semi-automated.

**Comment by elextr** [comment]
> AFAIK the styles are just compile-time constants so it's not obvious how they could be stored in the filetype file. Perhaps Scintilla provides some other way too, IDK.

Well themes set styles using names for the lexical entity, my (not fleshed out) thought at the time of the comment was that the same mechanism could be used to translate names to style numbers?

> More generally it would be good if we could auto-generate C code based on scintilla styles so adding a lexer is semi-automated.

Agree totally, IIRC @codebrainz once looked at scripting Lexer upgrades further, but (again IIRC) when he asked for changes to Scintilla Neil refused (backward compatibility) so the idea was dropped.  I think some other Scintilla users have such tooling, thats why Neil didn't want to change stuff, maybe we could ~steal~ reuse it.
